plugins/primus_lisp/semantics/bits.lisp

```diff
@@ -19,13 +19,87 @@
          (logand (msb rm) (lnot (msb rd)))
          (logand (msb rn) (lnot (msb rd)))))
 
-
 (defun overflow (rd rn rm)
   "(overflow RD RN RM) is true if the sum RD = RN + RM results
    in two's complement overflow."
   (logor (logand (msb rn) (msb rm) (lnot (msb rd)))
          (logand (lnot (msb rn)) (lnot (msb rm)) (msb rd))))
 
+(defun highest-set-bit (bitv)
+  "(highest-set-bit bitv) returns the greatest index whose bit is set in bitv.
+   It requires bitv to be non-zero.
+   Translated from ARMv8 ISA pseudocode."
+  (assert-msg (not (is-zero bitv)) "highest-set-bit bitv is zero")  ; at least 1 bit must be set
+  (let ((i (- (word-width bitv) 1)))
+    (while (and (> i 0) (= (select i bitv) 0))
+      (decr i))
+    i))
+
+(defun replicate (bitv n)
+  "(replicate bitv n) returns a bitvector with bitv repeated n times.
+   Translated from ARMv8 ISA pseudocode."
+  (let ((output 0:0))
+    (while (> n 0)
+      (decr n)
+      (set output (concat output bitv)))
+    output))
+
+(defun replicate-to-fill (bitv n)
```

Comment:
I think that this function is too specific and should be moved to the arm-bits.lisp. We already have `replicate` and this function is just a specific wrapper around it that makes sense for implementing arm instructions.

Reply:
ok